File: frontend_lang/src/types/course.ts
export interface Course {
  id: number;
  targetLanguage?: string;
  language: string;
  category?: string;
  title: string;
  description: string;
  japaneseTag?: string;
  totalLessons: number;
  freeLessonsCount: number;
}

export interface CreateCourseData {
  targetLanguage?: string;
  category: string;
  title: string;
  description: string;
  japaneseTag?: string;
}

export interface CreateLessonData {
  title: string;
  videoUrl: string;
  isFree: boolean;
  sequenceNo?: number;
}

File: backend_lang/src/main/java/com/model/backend_lang/model/Course.java
package com.model.backend_lang.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "courses")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Course {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String language;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "japanese_tag")
    private String japaneseTag;

    @OneToMany(mappedBy = "course", cascade = CascadeType.ALL, orphanRemoval = true)
    @OrderBy("sequenceNo ASC")
    @Builder.Default
    private List<Lesson> lessons = new ArrayList<>();
}
